### DISCUSSION TITLE
fastapi Request to multidict for wtforms error

### DISCUSSION THREAD

**Discussion by imiten** [encode/starlette#1585] [Q&A]
Hi,

Here is snippet with only relevant portions mentioned.  I am trying to use WTForms with FastAPI to get form validation for HTML based web app.  We do not use json as mime type for rest instead use HTML only.  

for last line in code spinnet where I instantiate GameForm I get error.  I have tried passing request, request.form(), request.body() etc.
how do I pass fastapi Request to wftforms Form object. I am getting error: TypeError: formdata should be a multidict-type wrapper that supports the 'getlist' method.

snippet:

```python
from wtforms  import Form, StringField, TextAreaField
from wtforms.validators import ValidationError
.....
class GameForm(Form):
    name = StringField("Game Name")
    json_string = TextAreaField("JSON")

    def validate_json_string(form, field):
        try:
            json.dumps(field.data)
        except:
            raise ValidationError("invalid json")
.....

@app.post("/game", response_class=HTMLResponse)
def create_game(request: Request, 
         db: Session = Depends(get_db)):
    """
      Add new game and stores in database
    """
    form = GameForm(request.form())    
```

Regards,

Miten.

**Comment by Kludex** [comment]
Please provide a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), with the version of the dependencies being used.

**Comment by imiten** [comment]
[games.zip](https://github.com/encode/starlette/files/8479857/games.zip)

**Reply by imiten** [reply]
I was able to use request.form() and pass it as formdata.  had to use async await.  Also for validating json I used loads instead of dumps (which was used when pydantic was used).  here is snippet:

```
@app.post("/game", response_class=HTMLResponse)
async def create_game(request: Request, 
         db: Session = Depends(get_db)):
    """
      Add new game and stores in database
    """
    formdata = await request.form()
    form = GameForm(formdata=formdata)    

```

still is would be ideal that we can pass just the request may be it will be more flexible and inline with existing solutions.
